Repository/PhpcrOdmRepository.php

```diff
@@ -116,4 +128,62 @@ protected function buildCollection(array $documents)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::notEq('', trim($path, '/'), 'You cannot create a root node.');
+        Assert::startsWith($path, '/', 'Target path "%s" must be absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentPath = PathHelper::getParentPath($resolvedPath);
+        $parentDocument = $this->getManager()->find(null, $parentPath);
+
+        if (null === $parentDocument) {
+            throw new InvalidArgumentException(sprintf('Cannot locate parent document at "%s"', $parentPath));
+        }
+
+        /** @var PhpcrOdmResource[] $resources */
+        $resources = $resource instanceof IteratorAggregate ? $resource : new ArrayResourceCollection([$resource]);
+        Assert::isInstanceOf($resources, ResourceCollection::class, 'The list should be of instance "ResourceCollection".');
+
+        foreach ($resources as $resource) {
+            Assert::isInstanceOf($resource, CmfResource::class, 'The resource needs to of instance "CmfResource".');
+            Assert::notNull($resource->getName(), 'The resource needs a name for the creation.');
+
+            $document = $resource->getPayload();
+            $document->setName($resource->getName());
+            if ($document instanceof HierarchyInterface) {
```

Comment:
Currently creation and edition would be/is done either by SonataAdmin or SyliusResourceBundle - which supports both HTML and REST endpoints for editing and creating new objects.

I see the Resource system as providing links to these endpoints which are capable of dealing with the resource's payload.